qiime2/core/archive/provenance.py

```diff
@@ -348,9 +347,11 @@ def make_action_section(self):
 
 
 class ActionProvenanceCapture(ProvenanceCapture):
-    def __init__(self, action_type, import_path, action_id):
+    def __init__(self, action_type, plugin_id, action_id):
+        from qiime2.sdk import PluginManager
+
         super().__init__()
-        self._plugin = importlib.import_module(import_path).__plugin__
+        self._plugin = PluginManager().get_plugin(id=plugin_id)
```

Comment:
this is so much less cludgy it's unreal

Reply:
ok, so the thing I want to test is if `concurrent.futures` does something particular odd about the memory mapping for Python modules (which would cause the plugin to be the same address) vs the new approach, which may create a new PluginManager, with potentially new state (such that the plugin that was added, is not available in the "copy" as it was never added in the clone).

Reply:
ok, empirically, this works fine, I created a PluginManager and added only the `dummy_plugin`, then did an async call on `parameter_only_pipelines` and got all of the provenance I expected, and the dummy_plugin was identified (which would be missing if we had a different PluginManager).

```pytb
In [1]: from qiime2.sdk import PluginManager                                                                                                  

In [2]: pm = PluginManager(add_plugins=False)                                                                                                 

In [3]: pm.plugins                                                                                                                            
Out[3]: {}

In [4]: from qiime2.core.testing.plugin import dummy_plugin                                                                                   

In [5]: pm.add_plugin(dummy_plugin)                                                                                                           

In [6]: pm.plugins                                                                                                                            
Out[6]: {'dummy-plugin': <qiime2.plugin.plugin.Plugin at 0x7fd29b753da0>}

In [7]: p = dummy_plugin.actions['parameter_only_pipeline']                                                                                   

In [8]: fut = p.asynchronous(1, 2)                                                                                                            

In [9]: r = fut.result()                                                                                                                      

In [10]: r                                                                                                                                    
Out[10]: 
Results (name = value)
-------------------------------------------------------------------------
foo = <artifact: IntSequence2 uuid: 4338de5f-ae26-44b5-b883-e8567d0168ef>
bar = <artifact: IntSequence1 uuid: 32c1b8c8-d385-4e5e-ae98-f2432d2d075e>
```